src/clr-angular/data/datagrid/providers/table-size.service.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {isPlatformBrowser} from "@angular/common";
+import {ElementRef, Inject, Injectable, PLATFORM_ID} from "@angular/core";
+
+/**
+ * @description
+ * Internal datagrid service that holds a reference to the clr-dg-table element and exposes a method to get height.
+ */
+@Injectable()
+export class TableSizeService {
+    private tableRef: Element;
+
+    constructor(@Inject(PLATFORM_ID) private platformId: Object) {}
+    public set table(table: ElementRef) {
+        if (isPlatformBrowser(this.platformId) && table.nativeElement) {
+            this.tableRef = table.nativeElement.querySelector(".datagrid-table");
+        }
+    }
+
+    // Used when resizing columns to show the column border being dragged.
+    getColumnDragHeight(): string {
+        return `${this.tableRef.clientHeight}px`;
```

Comment:
The DnD topic branch has DomAdapter changes. So once that's merged in, we could use it here.